tests/core/test_config.py

```diff
@@ -110,3 +111,118 @@ def testDefaultChecks(self):
+            '%s/test_check.py' % self.TEMP_AGENT_CHECK_DIR)
+        copyfile('%s/valid_check_1.py' % self.FIXTURE_PATH,
+            '%s/test_check.py' % self.TEMP_ETC_CHECKS_DIR)
+        checks = load_check_directory({"additional_checksd": self.TEMP_ETC_CHECKS_DIR}, "foo")
+        self.assertEquals(1, len(checks['initialized_checks']))
+        self.assertEquals('valid_check_1', checks['initialized_checks'][0].check(None))
+
+    def testConfigCheckNotAgentCheck(self):
+        copyfile('%s/valid_conf.yaml' % self.FIXTURE_PATH,
+            '%s/test_check.yaml' % self.TEMP_ETC_CONF_DIR)
+        copyfile('%s/invalid_check_1.py' % self.FIXTURE_PATH,
+            '%s/test_check.py' % self.TEMP_AGENT_CHECK_DIR)
+        checks = load_check_directory({"additional_checksd": self.TEMP_ETC_CHECKS_DIR}, "foo")
+        self.assertEquals(0, len(checks['init_failed_checks']))
+        self.assertEquals(0, len(checks['initialized_checks']))
+
+    def testConfigCheckImportError(self):
+        copyfile('%s/valid_conf.yaml' % self.FIXTURE_PATH,
+            '%s/test_check.yaml' % self.TEMP_ETC_CONF_DIR)
+        copyfile('%s/invalid_check_2.py' % self.FIXTURE_PATH,
+            '%s/test_check.py' % self.TEMP_AGENT_CHECK_DIR)
+        checks = load_check_directory({"additional_checksd": self.TEMP_ETC_CHECKS_DIR}, "foo")
+        self.assertEquals(1, len(checks['init_failed_checks']))
+
+    def testConfig3rdPartyAgent(self):
+        copyfile('%s/valid_conf.yaml' % self.FIXTURE_PATH,
+            '%s/test_check.yaml' % self.TEMP_ETC_CONF_DIR)
+        copyfile('%s/valid_check_2.py' % self.FIXTURE_PATH,
+            '%s/test_check.py' % self.TEMP_AGENT_CHECK_DIR)
+        copyfile('%s/valid_check_1.py' % self.FIXTURE_PATH,
+            '%s/test_check/check.py' % self.TEMP_3RD_PARTY_CHECKS_DIR)
+        checks = load_check_directory({"additional_checksd": self.TEMP_ETC_CHECKS_DIR}, "foo")
+        self.assertEquals(1, len(checks['initialized_checks']))
+        self.assertEquals('valid_check_1', checks['initialized_checks'][0].check(None))
+
+    def testConfigETC3rdParty(self):
+        copyfile('%s/valid_conf.yaml' % self.FIXTURE_PATH,
+            '%s/test_check.yaml' % self.TEMP_ETC_CONF_DIR)
+        copyfile('%s/valid_check_2.py' % self.FIXTURE_PATH,
+            '%s/test_check/check.py' % self.TEMP_3RD_PARTY_CHECKS_DIR)
+        copyfile('%s/valid_check_1.py' % self.FIXTURE_PATH,
+            '%s/test_check.py' % self.TEMP_ETC_CHECKS_DIR)
+        checks = load_check_directory({"additional_checksd": self.TEMP_ETC_CHECKS_DIR}, "foo")
+        self.assertEquals(1, len(checks['initialized_checks']))
+        self.assertEquals('valid_check_1', checks['initialized_checks'][0].check(None))
+
+    def tearDown(self):
+        import config as Config
+        Config.get_checksd_path = self.patched_get_checksd_path
+        Config.get_confd_path = self.patched_get_confd_path
+        Config.get_3rd_party_path = self.patched_get_3rd_party_path
```

Comment:
This pattern seems weird. Is it possible to use a `MockConfig` object with the dummy methods, and then decorate at the class-level: 
```
@patch.object(config, 'Config', MockConfig)
class TestConfigLoadCheckDirectory(unittest.TestCase):
```